### PR TITLE
[relay-runtime] Correct typing for default missing field handlers

### PR DESCRIPTION
**Pull request by ethanve**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/relay/blob/master/packages/relay-runtime/handlers/RelayDefaultMissingFieldHandlers.js#L31

### CODE REVIEW

File: types/relay-runtime/lib/handlers/RelayDefaultMissingFieldHandlers.d.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import { MissingFieldHandler } from '../store/RelayStoreTypes';
 
-export const missingViewerFieldHandler: MissingFieldHandler;
```
Comment (ethanve) [inline types/relay-runtime/lib/handlers/RelayDefaultMissingFieldHandlers.d.ts]: `missingViewerFieldHandler` isn't being exported by this filer rather, tt's exporting an array with `missingViewerFieldHandler` as the only element. Not sure if this new typing is accurately representing that

Comment (alloy) [reply types/relay-runtime/lib/handlers/RelayDefaultMissingFieldHandlers.d.ts]: Good catch!

1. However, the upstream version doesn’t actually export a named type, rather a default…

File: types/relay-runtime/lib/handlers/RelayDefaultMissingFieldHandlers.d.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
...
+export const DefaultMissingFieldHandlers: MissingFieldHandler[];
```
Comment (alloy) [inline types/relay-runtime/lib/handlers/RelayDefaultMissingFieldHandlers.d.ts]: 2. …so this should really be something like:

```suggestion
declare const DefaultMissingFieldHandlers: MissingFieldHandler[];
export default DefaultMissingFieldHandlers
```

File: types/relay-runtime/index.d.ts [hunk 3]
```diff
@@ -170,9 +170,7 @@ export { readInlineData } from './lib/store/readInlineData';
 
 // Extensions
 export { RelayDefaultHandlerProvider as DefaultHandlerProvider } from './lib/handlers/RelayDefaultHandlerProvider';
-export {
-    missingViewerFieldHandler as DefaultMissingFieldHandlers,
-} from './lib/handlers/RelayDefaultMissingFieldHandlers';
+export { DefaultMissingFieldHandlers } from './lib/handlers/RelayDefaultMissingFieldHandlers';
```
Comment (alloy) [inline types/relay-runtime/index.d.ts]: 3. After which [the below suggested change] this should be:

```suggestion
import DefaultMissingFieldHandlers from './lib/handlers/RelayDefaultMissingFieldHandlers';
export { DefaultMissingFieldHandlers };
```

**Review by alloy — CHANGES_REQUESTED** [review]
Conceptually good, just some minor suggestions to make it follow upstream exactly 👍

**Comment by alloy** [comment]
Great, thanks!

**Pull request merged**